cli/running/watchdog.go

```diff
@@ -78,19 +75,20 @@ func (wd *Watchdog) Start() error {
 	// get started for avoiding a race condition between tt start
 	// and tt stop. This way we avoid a situation when we receive
 	// a signal before starting a handler for it.
-	wd.startSignalHandling()
+	goroutineCtx, goroutineCancel := context.WithCancel(context.Background())
+	wd.startSignalHandling(goroutineCtx, goroutineCancel)
 
 	// Launch integrity checking goroutine.
 	if wd.integrityCheckPeriod != 0 {
 		wd.logger.Printf("(INFO): starting periodic integrity checks each %s.",
 			wd.integrityCheckPeriod)
-		wd.startIntegrityChecks()
+		wd.startIntegrityChecks(goroutineCtx)
 	}
 
 	if err = wd.preStartAction(); err != nil {
 		wd.logger.Printf(`(ERROR): Pre-start action error: %v`, err)
-		// Finish the signal handling goroutine.
-		wd.done <- true
+		// Finish the signal handling and periodic integrity checking goroutines.
+		goroutineCancel()
```

Comment:
Do we need to wait goroutines here?
```suggestion
		goroutineCancel()
		wd.doneBarrier.Wait()
```

Reply:
We can and it doesn't seem to hurt, although I guess it's not really necessary -- it seems to me that the only place we need to synchronise with the goroutines is when we are restarting them, and we don't do this here.

Reply:
I've added the synchronization here anyway. Maybe it will save us some pain in the future.